File: samples/python/logstream/stream.py
import asyncio
import sys
import threading
import time
import websockets
import requests
import json


async def connect(url):
    async with websockets.connect(url, subprotocols=['json.webpubsub.azure.v1']) as ws:
        print('connected')
        id = 1
        while True:
            data = input()
            payload = {
                'type': 'sendToGroup',
                'group': 'stream',
                'dataType': 'text',
                'data': str(data),
                'ackId': id
            }
            id = id + 1
            await ws.send(json.dumps(payload))
            await ws.recv()

res = requests.get('http://localhost:8080/negotiate').json()

try:
    asyncio.get_event_loop().run_until_complete(connect(res['url']))
except KeyboardInterrupt:
    pass

File: samples/javascript/logstream/stream.js
const WebSocket = require('ws');
const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));

async function main() {
  let res = await fetch(`http://localhost:8080/negotiate`);
  let data = await res.json();
  let ws = new WebSocket(data.url, 'json.webpubsub.azure.v1');
  let ackId = 0;
  ws.on('open', () => {
    process.stdin.on('data', data => {
      let a = JSON.stringify({
        type: 'sendToGroup',
        group: 'stream',
        dataType: 'text',
        ackId: ++ackId,
        data: data.toString()
      });
      ws.send(a);
    });
  });
  ws.on('message', data => console.log("Received: %s", data));
  process.stdin.on('close', () => ws.close());
}

main();
